src/shared.rs

```diff
@@ -109,7 +105,12 @@ impl Address {
     }
     /// This creates a new address from a value that implements [std::string::ToString]
     pub fn new<T: ToString>(string: T) -> Self {
-        Self(string.to_string())
+        let str = string.to_string();
+        if str.starts_with("0x") {
+            Self(str)
+        } else {
+            Self("0x".to_owned() + str.as_str())
```

Comment:
Use format!

Reply:
Format is actually slower in this case, the original code way back when, used `format!`

Reply:
would probs be wise to stick a comment on it saying that so nobody tries to refactor it in the future